core/src/main/java/cucumber/runtime/formatter/JSONFormatter.java

```diff
@@ -382,14 +384,14 @@ private void addEmbeddingToHookMap(byte[] data, String mimeType) {
         if (result.getErrorMessage() != null) {
             resultMap.put("error_message", result.getErrorMessage());
         }
-        if (result.getDuration() != null && result.getDuration() != 0) {
-            resultMap.put("duration", result.getDuration());
+        if (result.getDuration() != 0) {
+            resultMap.put("duration", MILLISECONDS.toNanos(result.getDuration()));
```

Comment:
Do we still need nanos?

We have lost the precision of nanos anyway, and `toNanos()` basically scales it by **1000000** i think. We could just use millis if there isnt a specific reason for using nanos.

Reply:
Yes. The consumers of the JSON report would still expect the time to be expressed in nano seconds.

Reply:
We could deprecate it to allow consumers to migrate, if that seems feasible.

Reply:
I've reverted the use of milliseconds for duration. Individual steps are still quite close to 1ms.